File: patch_taobao.js
const fs = require('fs');
const path = 'pages-functions/functions/api/_taobao.ts';
let c = fs.readFileSync(path, 'utf8');

// Fix 1: Increase MIN_PRICE from 30 to 50
c = c.replace('const MIN_PRICE = 30;', 'const MIN_PRICE = 50;');

// Fix 2: Add sample filtering to second pass
const oldSecondPass = `  // Second pass: best-available within price range
  for (var i = 0; i < products.length; i++) {
    if (products[i].price >= MIN_PRICE && products[i].price <= MAX_PRICE) {
      console.log("[taobao] Best avail: " + products[i].title.substring(0, 40) + " (¥" + products[i].price + ")");
      return products[i];
    }
  }`;

const newSecondPass = `  // Second pass: best-available within price range, also filter samples
  for (var i = 0; i < products.length; i++) {
    if (products[i].price < MIN_PRICE || products[i].price > MAX_PRICE) continue;
    var sampleKws2 = ["小样", "试用装", "中样", "体验装", "1ml", "2ml", "3ml", "5ml", "10ml", "0.5ml", "1.5ml", "30ml*3", "3支装", "5支装", "支装", "分装"];
    if (sampleKws2.some(function(s) { return products[i].title.indexOf(s) !== -1; })) continue;
    console.log("[taobao] Best avail: " + products[i].title.substring(0, 40) + " (¥" + products[i].price + ")");
    return products[i];
  }`;

if (c.includes(oldSecondPass)) {
  c = c.replace(oldSecondPass, newSecondPass);
  console.log('Second pass patched successfully');
} else {
  console.log('Second pass pattern not found, trying alternative...');
  // Try line by line
  const lines = c.split('\n');
  let found = false;
  for (let i = 0; i < lines.length; i++) {
    if (lines[i].trim() === '// Second pass: best-available within price range') {
      console.log('Found at line', i + 1);
      found = true;
      break;
    }
  }
  if (!found) console.log('Pattern not found at all');
}

fs.writeFileSync(path, c);
console.log('File written, size:', c.length);

File: patch_taobao.py
import re

path = r"C:\Users\yao\Documents\ChatGPT\美妆app\pages-functions\functions\api\_taobao.ts"
with open(path, "r", encoding="utf-8") as f:
    content = f.read()

# Add MIN_PRICE filter constant after CACHE_TTL
content = content.replace(
    "const CACHE_TTL = 24 * 60 * 60;",
    "const CACHE_TTL = 24 * 60 * 60;\nconst MIN_PRICE = 20; // 低于20元的商品视为小样/劣质品，跳过"
)

# Replace findProductByKeyword to add price filtering
old_func = """export async function findProductByKeyword(productName: string, env: Ctx['env']): Promise<TaobaoProduct | null> {
  if (!productName || productName.trim().length < 2) return null;
  var products = await searchTaobaoProducts(productName.trim(), env, 10);
  if (products.length === 0) return null;
  var name = productName.trim();
  var matched = products.find(function(p) { return p.title.includes(name) || name.includes(p.title.substring(0, 4)); });
  return matched || products[0];
}"""

new_func = """export async function findProductByKeyword(productName: string, env: Ctx['env']): Promise<TaobaoProduct | null> {
  if (!productName || productName.trim().length < 2) return null;
  var products = await searchTaobaoProducts(productName.trim(), env, 10);
  if (products.length === 0) return null;
  var name = productName.trim();
  var matched = products.find(function(p) { return p.title.includes(name) || name.includes(p.title.substring(0, 4)); });
  if (!matched) return products[0];
  // Filter out samples/cheap goods: skip if price < MIN_PRICE or title contains 小样/试用装/中样
  if (matched.price < MIN_PRICE) {
    console.log('[taobao] Price filter skipped: ' + matched.title + ' (¥' + matched.price + ')');
    return null;
  }
  var sampleKeywords = ['小样', '试用装', '中样', '体验装', '10ml', '5ml', '3ml'];
  for (var i = 0; i < sampleKeywords.length; i++) {
    if (matched.title.indexOf(sampleKeywords[i]) !== -1) {
      console.log('[taobao] Sample filter skipped: ' + matched.title);
      return null;
    }
  }
  return matched;
}"""

content = content.replace(old_func, new_func)

with open(path, "w", encoding="utf-8") as f:
    f.write(content)
print("Patched _taobao.ts")
